packages/salesforcedx-vscode-core/test/vscode-integration/util/inputUtils.test.ts

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2022, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { assert, expect } from 'chai';
+import { createSandbox, SinonSandbox, SinonStub } from 'sinon';
+import * as vscode from 'vscode';
+import { getTrimmedString } from '../../../src/util/inputUtils';
+
+describe('inputUtils Unit tests', () => {
+  const sandbox = createSandbox();
+  const showInputBoxStub = sandbox.stub(vscode.window, 'showInputBox');
+
+  describe('getTrimmedString', () => {
+    const INPUT_VAL = 'Test Input';
+    const EMPTY_STRING = '';
+    const WHITESPACE = '   ';
+
+    after(() => {
+      sandbox.restore();
+    });
+
+    it('should call showInputBox once', async () => {
+      showInputBoxStub.resolves(INPUT_VAL);
+      const trimmedString = await getTrimmedString({});
+      assert(showInputBoxStub.calledOnce);
+    });
+
+    it('should remove leading whitespace', async () => {
+      showInputBoxStub.resolves(`  ${INPUT_VAL}`);
+      const trimmedString = await getTrimmedString({});
+      expect(trimmedString).to.be.eq(INPUT_VAL);
+    });
+
+    it('should remove trailing whitespace', async () => {
+      showInputBoxStub.resolves(`${INPUT_VAL}  `);
+      const trimmedString = await getTrimmedString({});
+      expect(trimmedString).to.be.eq(INPUT_VAL);
+    });
+
+    it('should remove leading and trailing whitespace', async () => {
+      showInputBoxStub.resolves(`  ${INPUT_VAL}  `);
+      const trimmedString = await getTrimmedString({});
+      expect(trimmedString).to.be.eq(INPUT_VAL);
+    });
+
+    it('should return an empty string when given an empty string', async () => {
+      showInputBoxStub.resolves(EMPTY_STRING);
+      const trimmedString = await getTrimmedString({});
+      expect(trimmedString).to.be.eq(EMPTY_STRING);
+    });
+
+    it('should return empty string when given whitespace', async () => {
+      showInputBoxStub.resolves(WHITESPACE);
+      const trimmedString = await getTrimmedString({});
+      expect(trimmedString).to.be.eq(EMPTY_STRING);
+    });
+
+    it('should return undefined when given undefined', async () => {
+      showInputBoxStub.resolves(undefined);
+      const trimmedString = await getTrimmedString({});
+      assert.isUndefined(trimmedString);
+    });
+
+    it('should return null when given null', async () => {
```

Comment:
Since getTrimmedString uses showInputBox internally, which only returns a string or undefined, this might not be a valid test.  Thoughts?

Reply:
@klewis-sfdc - Sorry for looking at this so late. Instead of getTrimmedString taking in InputBoxOptions, if it just takes a string (whatever vscode.window.showInputBox() returns) and trims it, would the tests and the test changes be simpler?

Reply:
Yes, it would definitely be simpler, but there was some discussion in the first few rounds of review on this PR about having a reusable service method that could be used anytime we wanted to get a trimmed string from the user to avoid having to do it _after_ collection of the value each time.  I'm ok either way and could update the PR to just trim before returning from SelectProjectName (and removing getTrimmedString) if you think that is better.  

Alternatively:
<img width="1411" alt="2022-04-04_13-19-16" src="https://user-images.githubusercontent.com/46458081/161625813-129e5dc8-b8fa-44ed-b531-04c20a2380e7.png">

Reply:
It does seem a bit weird passing input options to the getTrimmedString function, but if we switch to passing it a String, then probably better to just call string.trim() locally here.

Reply:
Agree, just calling string.trim() would be my preference as well. I see the point about a reusable function - but this is such a simple cleanup on the string vs a validation for function name or apex class name.

Reply:
We chatted in slack to coordinate. Ended up keeping the utility method as it was already mostly there.